### ISSUE TITLE
cannot push multiple types onto array in new type inference

### ISSUE THREAD

**Issue by nicks** [google/closure-compiler#1455]
Code

``` js
var array = []
array.push(false, 1)
```

Expected: No warnings:
Actual:

```
input0:2: WARNING - Illegal instantiation in function type function(this:Array<T>|{length:number}|null,...T):number for the type variable T.
 You can only specify one type. Found [boolean, number].
array.push(false, 1)
^
```

http://closure-compiler-debugger.appspot.com/#input0%3Dvar%2520array%2520%253D%2520%255B%255D%250Aarray.push(false%252C%25201)%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CHECK_TYPES_NEW_INFERENCE%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1
